### PR TITLE
Document the release process

### PR DESCRIPTION
**Pull request by vcanales**
Adds a description of the release process.

Tasks:

- [x] Provide clarity on who is actually able to review and merge.
- [x] Warn against closing release PRs without merging.

### CODE REVIEW

**Comment by pbking** [comment]
> Use the `git cherry-pick [commit-hash]` command to add specific commits to the Release Branch associated with the Release PR.

Does the changelog continue to get updated or does that need to happen manually with cherry-picked branches?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -50,6 +50,24 @@ There are several linter commands available to help ensure the plugin follows th
...
+
+
+### 2 - Update the Release PR
+
+Keep the Release PR current by incorporating any new changes from the `trunk` that are intended for this release. Use the `git cherry-pick [commit-hash]` command to add specific commits to the Release Branch associated with the Release PR. The Release Branch is named using the format: `release/[creation-date]/[release-type]-release`, where `[creation-date]` is the date the Release PR was created, and `[release-type]` is the type selected during the workflow initiation.
+
+
+### 3 - Finalize the Release
+
+Once the release is deemed complete and ready, it must be reviewed and approved by members of the organization. Following approval, the Release PR is merged into the main branch. This action triggers the [**Deploy to Dotorg**](https://github.com/WordPress/create-block-theme/actions/workflows/deploy-to-dotorg.yml) workflow, which tags the release on both GitHub and the WordPress Plugin Directory SVN. A Release Confirmation is then triggered on WordPress.org, notifying plugin maintainers via email of the new release awaiting confirmation. Upon confirmation, the new version becomes live on the WordPress Plugin Directory.
```
Comment (pbking) [inline CONTRIBUTING.md:69]: Does github display the members in this list?

**Comment by vcanales** [comment]
> > Use the `git cherry-pick [commit-hash]` command to add specific commits to the Release Branch associated with the Release PR.
> 
> Does the changelog continue to get updated or does that need to happen manually with cherry-picked branches?

Good point, I'll need to add a step that automatically updates the Change Log after new commits are added.

**Review by pbking — APPROVED** [review]
A good change for now.  Mentioned cherry-picked changes after creation which will need a follow up.

🚢

**Pull request merged**